tests/python/pants_test/backend/codegen/tasks/test_protobuf_integration.py

```diff
@@ -15,6 +15,12 @@
 
 class ProtobufIntegrationTest(PantsRunIntegrationTest):
 
+  def test_import_from_buildroot(self):
+    pants_run = self.run_pants(
+      ['gen.protoc', '--import-from-root',
+       'testprojects/src/protobuf/org/pantsbuild/testproject/import_from_buildroot/bar'])
+    self.assert_success(pants_run)
+
```

Comment:
It wouldn't fail in that cast, as far as I can tell. And even before, you were free to import any proto file in any source root that was in your other dependencies' `bases` (or your own), even if you didn't depend on it. 

This is true in all codegen, I think, and in many other tasks.  But I'd rather we put energy into moving in the other direction - namely implying dependencies when we can instead of requiring them to be explicit... :)